dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java

```diff
@@ -127,9 +128,25 @@ public String getStringKey() {
         private final List<byte[]> binaryKeys;
         private final OpName op;
 
-        private MultiKeyOperation(final List<String> keys, final OpName o) {
-            this.keys = keys;
-            this.binaryKeys = null;
+        private MultiKeyOperation(final List keys, final OpName o) {
+            Object firstKey = (keys != null) ? keys.get(0) : null;
+
+            if(firstKey != null) {
+                if (firstKey instanceof String) {//string key
+                    this.keys = keys;
+                    this.binaryKeys = null;
+                } else if (firstKey instanceof byte[]) {//binary key
+                    this.keys = null;
+                    this.binaryKeys = keys;
+                } else {//something went wrong here
+                    this.keys = null;
+                    this.binaryKeys = null;
+                }
+            } else {
+                this.keys = null;
```

Comment:
Is there a case that the `firstKey` is `null` and the rest are not?

Reply:
@ipapapa  good point. I have addd the null filter to address it and when all nulls sent it fails. Even the old implementation had this issue.

Reply:
@ipapapa  Even when we filter nulls the change for this is across all places where ever we use MultiKeyOperation and CompressionValueMultiKeyOperation and it will fail saying 

`redis.clients.jedis.exceptions.JedisDataException: value sent to redis cannot be null`

so its a known  behavior even with current implementation. If you want I can go ahead and add a Collections filter on all places before passing the keys down further. Please let me know.